core/src/components/searchbar/searchbar.tsx

```diff
@@ -76,11 +77,12 @@ export class Searchbar implements ComponentInterface {
   /**
    * Set the amount of time, in milliseconds, to wait to trigger the `ionInput` event after each keystroke.
    */
-  @Prop() debounce = 250;
+  @Prop() debounce?: number;
 
   @Watch('debounce')
   protected debounceChanged() {
-    this.ionInput = debounceEvent(this.ionInput, this.debounce);
+    const { ionInput, debounce, unmoddedIonInput } = this;
+    this.ionInput = debounce === undefined ? unmoddedIonInput : debounceEvent(ionInput, debounce);
```

Comment:
Is there a challenge we ran into that led to a private variable to track the event vs.:
```ts
const { debounce } = this;
if (debounce !== undefined) {
   this.ionInput = debounceEvent(this.ionInput, debounce);
}
```

Reply:
Doing it that way would cause `ionInput` to stay debounced if you dynamically changed `debounce` from some number to `undefined`.

Reply:
Ahh that makes sense, can we add a note about that somewhere?

Also nit: We use `original` to prefix variables in [a few places](https://github.com/ionic-team/ionic-framework/search?q=original), should we follow that naming strategy here? e.g.: `originalIonInput`

Reply:
Done 👍